cpp/src/arrow/util/cpu_info.cc

```diff
@@ -59,17 +59,75 @@ void __cpuidex(int CPUInfo[4], int function_id, int subfunction_id) {
 }
 #endif
 
+#if defined(__GNUC__) && defined(__linux__) && defined(__aarch64__)
+// There is no direct instruction to get cache size on Arm64 like '__cpuid' on x86;
+// Get Arm64 cache size by reading '/sys/devices/system/cpu/cpu0/cache/index*/size';
+// index* :
+//   index0: L1 Dcache
+//   index1: L1 Icache
+//   index2: L2 cache
+//   index3: L3 cache
+#include <errno.h>
+bool get_cache_size_arm64(const char* filename, int* size) {
+  bool cache_supported = false;
+  char* content = nullptr;
+  char* last_char = nullptr;
+  size_t file_len = 0;
+  int cardinal_num = 0;
+  int multip = 0;
+
+  if (size == nullptr) return cache_supported;
+  *size = 0;
+
+  FILE* cache_file = fopen(filename, "r");
+  if (cache_file == nullptr) return cache_supported;
+
+  // Read cache file to 'content' for getting cache size.
+  // Filename: '/sys/devices/system/cpu/cpu0/cache/index0/size' -> 32k
+  if (getline(&content, &file_len, cache_file) == -1) goto done;
+
+  errno = 0;
+  cardinal_num = strtoull(content, &last_char, 0);
+  if (errno != 0) goto done;
+
+  // KB, MB, or GB
+  multip = 1;
+  switch (*last_char) {
+    case 'g':
+    case 'G':
+      multip = 1024;
+    case 'm':
+    case 'M':
+      multip = multip * 1024;
+    case 'k':
+    case 'K':
+      multip = multip * 1024;
+  }
+  *size = cardinal_num * multip;
+  cache_supported = true;
+
+done:
+  fclose(cache_file);
+  free(content);
+
+  return cache_supported;
+}
+#endif
+
 namespace arrow {
 namespace internal {
 
 static struct {
   std::string name;
   int64_t flag;
 } flag_mappings[] = {
-    {"ssse3", CpuInfo::SSSE3},
-    {"sse4_1", CpuInfo::SSE4_1},
-    {"sse4_2", CpuInfo::SSE4_2},
-    {"popcnt", CpuInfo::POPCNT},
+#if (defined(_M_AMD64) || defined(_M_X64))
```

Comment:
I don't think this is sufficient, see here: https://sourceforge.net/p/predef/wiki/Architectures/